lib/hawkular/operations/operations_api.rb

```diff
@@ -215,7 +215,10 @@ def check_pre_conditions(hash = {}, params = [], &callback)
       fail 'callback must have the perform method defined. include Hawkular::Operations' unless
           callback.nil? || callback.respond_to?('perform')
       params.each do |property|
-        fail "Hash property #{property} must be specified" if hash[property].nil?
+        if hash[property].nil?
+          error_message = "Hash property #{property} must be specified"
+          callback.nil? ? fail(ArgumentError, error_message) : callback.perform(:failure, error_message)
```

Comment:
Yes. It should only fail if there is no error callback at all


Am 12. Juli 2016 5:17:49 nachm. schrieb Ruben Vargas 
<notifications@github.com>:

>> @@ -215,7 +215,10 @@ def check_pre_conditions(hash = {}, params = [], 
>> &callback)
>>        fail 'callback must have the perform method defined. include 
>>        Hawkular::Operations' unless
>>            callback.nil? || callback.respond_to?('perform')
>>        params.each do |property|
>> -        fail "Hash property #{property} must be specified" if 
>> hash[property].nil?
>> +        if hash[property].nil?
>> +          error_message = "Hash property #{property} must be specified"
>> +          callback.nil? ? fail(ArgumentError, error_message) : 
>> callback.perform(:failure, error_message)
>
> Just for clarify, this should fail and throw an exception only in the 
> following cases:
>
> ```
> client.invoke_generic_operation(noop) do |on|
>   on.success do |_data|
>     puts 'This should have failed'
>   end
> end
> ```
> `client.invoke_generic_operation(noop)
> `
>
> but shouldn't fail in this case:
> ```
> client.invoke_generic_operation(noop) do |on|
>   on.success do |_data|
>     puts 'This should have failed'
>   end
>
>   on.failure do |error|
>     puts error
>   end
> end
>
> ```
> Is that correct? If this is the case, I'll do the changes.
>
>
>
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/hawkular/hawkular-client-ruby/pull/105/files/d566c480dcf622eb55326fb2742eed1b65964569#r70459461